### ISSUE TITLE
DD operator does not start on EKS 1.25

### ISSUE THREAD

**Issue by ij3net** [DataDog/datadog-operator#734]
We have same issue.
`EKS: 1.25`
Helm Chart Version
`Datadog-operator: 0.9.2`
```
{"level":"ERROR","ts":"2023-03-10T17:33:35Z","logger":"setup","msg":"Problem running manager","error":"failed to wait for datadogagent caches to sync: no matches for kind \"PodDisruptionBudget\" in version \"
policy/v1beta1\""}
```

_Originally posted by @ij3net in https://github.com/DataDog/helm-charts/issues/941#issuecomment-1464144074_

**Comment by ij3net** [comment]
`EKS: 1.25`
Helm Chart Version `0.9.2`
Datadog-operator version: `v1.0.0-rc.10`

```
{"level":"INFO","ts":"2023-03-10T18:06:44Z","logger":"setup","msg":"Version: v1.0.0-rc.10"}
{"level":"INFO","ts":"2023-03-10T18:06:44Z","logger":"setup","msg":"Build time: 2023-03-01/23:46:07"}
{"level":"INFO","ts":"2023-03-10T18:06:44Z","logger":"setup","msg":"Git Commit: 64bcaf5e7732fe7e11cd2f0ce70c790939c3a782"}
{"level":"INFO","ts":"2023-03-10T18:06:44Z","logger":"setup","msg":"Go Version: go1.18.10"}
{"level":"INFO","ts":"2023-03-10T18:06:44Z","logger":"setup","msg":"Go OS/Arch: linux/amd64"}
{"level":"INFO","ts":"2023-03-10T18:06:44Z","logger":"setup","msg":"Manager will be watching namespace","namespace":""}
{"level":"INFO","ts":"2023-03-10T18:06:45Z","logger":"klog","msg":"Waited for 1.039305137s due to client-side throttling, not priority and fairness, request: GET:https://172.20.0.1:443/apis/telemetry.istio.io/v1alpha1?timeout=32s\n"}
{"level":"INFO","ts":"2023-03-10T18:06:46Z","logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8383"}
{"level":"INFO","ts":"2023-03-10T18:06:47Z","logger":"setup","msg":"starting manager"}
{"level":"INFO","ts":"2023-03-10T18:06:47Z","msg":"Starting server","path":"/metrics","kind":"metrics","addr":":8383"}
{"level":"INFO","ts":"2023-03-10T18:06:47Z","msg":"Starting server","kind":"health probe","addr":":8081"}
{"level":"INFO","ts":"2023-03-10T18:06:47Z","logger":"klog","msg":"attempting to acquire leader lease datadog/datadog-operator-lock...\n"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"klog","msg":"successfully acquired lease datadog/datadog-operator-lock\n"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v2alpha1.DatadogAgent"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.Secret"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.ConfigMap"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.DaemonSet"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.Deployment"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogmonitor","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor","source":"kind source: *v1alpha1.DatadogMonitor"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogmonitor","msg":"Starting Controller","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor"}
{"level":"DEBUG","ts":"2023-03-10T18:07:50Z","logger":"events","msg":"Normal","object":{"kind":"ConfigMap","namespace":"datadog","name":"datadog-operator-lock","uid":"ebcc1962-a265-4785-af2e-39b7505775db","apiVersion":"v1","resourceVersion":"898359568"},"reason":"LeaderElection","message":"datadog-operator-65f5b964bc-flglw_23f70cc6-36ee-4050-9115-bab77e167b26 became leader"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.Role"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.RoleBinding"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.ServiceAccount"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.PodDisruptionBudget"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.NetworkPolicy"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.ClusterRole"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.ClusterRoleBinding"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"controller.datadogagent","msg":"Starting Controller","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"DatadogMetricForwarders","msg":"New Datadog metrics forwarder registred","ID":"datadog/datadog"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"DatadogMetricForwarders","msg":"Starting Datadog metrics forwarder","CustomResource.Namespace":"datadog","CustomResource.Name":"datadog"}
{"level":"INFO","ts":"2023-03-10T18:07:50Z","logger":"DatadogMetricForwarders","msg":"Got Datadog Site","CustomResource.Namespace":"datadog","CustomResource.Name":"datadog","site":"https://api.datadoghq.com"}
{"level":"ERROR","ts":"2023-03-10T18:07:50Z","logger":"klog","msg":"Observed a panic: \"invalid memory address or nil pointer dereference\" (runtime error: invalid memory address or nil pointer dereference)\ngoroutine 836 [running]:\nk8s.io/apimachinery/pkg/util/runtime.logPanic({0x195e040?, 0x2ce1c60})\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/runtime/runtime.go:74 +0x86\nk8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x73?})\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/runtime/runtime.go:48 +0x75\npanic({0x195e040, 0x2ce1c60})\n\t/usr/local/go/src/runtime/panic.go:838 +0x207\ngithub.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).getCredentialsV2(0xc0002fc810?, 0xc0002fc6e0)\n\t/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:611 +0x69\ngithub.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).setupV2(0xc000ae8e00)\n\t/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:261 +0x24f\ngithub.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).connectToDatadogAPI(0xc000ae8e00)\n\t/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:303 +0x58\nk8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x18, 0xc00007e400})\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:220 +0x1b\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x1e887b0?, 0xc000485800?}, 0xc00061ddd0?)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:233 +0x57\nk8s.io/apimachinery/pkg/util/wait.poll({0x1e887b0, 0xc000485800}, 0x10?, 0xf4a905?, 0xc0009b4360?)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:580 +0x38\nk8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext({0x1e887b0, 0xc000485800}, 0x10?, 0xc00007e400?)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:545 +0x49\nk8s.io/apimachinery/pkg/util/wait.PollImmediateUntil(0xc000ae8e00?, 0xc000924510?, 0x4be0ff?)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:536 +0x7c\ngithub.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).start(0xc000ae8e00, 0xc0002ad778?)\n\t/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:165 +0x10c\ncreated by github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*ForwardersManager).Register\n\t/workspace/pkg/controller/utils/datadog/forwarders_manager.go:69 +0x365\n"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13e5029]

goroutine 836 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x73?})
    /go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x195e040, 0x2ce1c60})
    /usr/local/go/src/runtime/panic.go:838 +0x207
github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).getCredentialsV2(0xc0002fc810?, 0xc0002fc6e0)
    /workspace/pkg/controller/utils/datadog/metrics_forwarder.go:611 +0x69
github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).setupV2(0xc000ae8e00)
    /workspace/pkg/controller/utils/datadog/metrics_forwarder.go:261 +0x24f
github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).connectToDatadogAPI(0xc000ae8e00)
    /workspace/pkg/controller/utils/datadog/metrics_forwarder.go:303 +0x58
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x18, 0xc00007e400})
    /go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:220 +0x1b
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x1e887b0?, 0xc000485800?}, 0xc00061ddd0?)
    /go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:233 +0x57
k8s.io/apimachinery/pkg/util/wait.poll({0x1e887b0, 0xc000485800}, 0x10?, 0xf4a905?, 0xc0009b4360?)
    /go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:580 +0x38
k8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext({0x1e887b0, 0xc000485800}, 0x10?, 0xc00007e400?)
    /go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:545 +0x49
k8s.io/apimachinery/pkg/util/wait.PollImmediateUntil(0xc000ae8e00?, 0xc000924510?, 0x4be0ff?)
    /go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:536 +0x7c
github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).start(0xc000ae8e00, 0xc0002ad778?)
    /workspace/pkg/controller/utils/datadog/metrics_forwarder.go:165 +0x10c
created by github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*ForwardersManager).Register
    /workspace/pkg/controller/utils/datadog/forwarders_manager.go:69 +0x365
Stream closed EOF for datadog/datadog-operator-65f5b964bc-flglw (datadog-operator)

```

**Comment by levan-m** [comment]
@ij3net, PDB compatibility issue with Kubernetes 1.25 has been addressed in v1.0 RC7 and later. See more details in https://github.com/DataDog/datadog-operator/issues/619 and https://github.com/DataDog/datadog-operator/issues/689. 

You can install v1.0, however keep in mind that this setup uses `DatadogAgent` version `v2alpha1` instead of `v1alpha1`.   Without webhook enabled (coming with GA) you won't be able to apply your existing `v1alpha1` manifests. You'll need to manually convert them to `v2alpha1`

`helm install datadog-operator datadog/datadog-operator --set image.tag=1.0.0-rc.10 --set datadog-crds.migration.datadogAgents.version=v2alpha1`

I'll resolve this issue since this question has been addressed in referenced ones. Please comment on #619 if you have any follow-up questions, or re-open if there are some EKS-specific nuances we may not be aware of.

**Comment by BrokenFlame** [comment]
The workaround throws new errors:
```sh
{"level":"ERROR","ts":"2023-03-19T07:35:53Z","logger":"klog","msg":"Observed a panic: \"invalid memory address or nil pointer dereference\" (runtime error: invalid memory address or nil pointer dereference)\ngoroutine 666 [running]:\nk8s.io/apimachinery/pkg/util/runtime.logPanic({0x195e040?, 0x2ce1c60})\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/runtime/runtime.go:74 +0x86\nk8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x73?})\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/runtime/runtime.go:48 +0x75\npanic({0x195e040, 0x2ce1c60})\n\t/usr/local/go/src/runtime/panic.go:838 +0x207\ngithub.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).getCredentialsV2(0xc0003acc30?, 0xc0003acb00)\n\t/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:611 +0x69\ngithub.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).setupV2(0xc000437dc0)\n\t/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:261 +0x24f\ngithub.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).connectToDatadogAPI(0xc000437dc0)\n\t/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:303 +0x58\nk8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x18, 0xc000680800})\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:220 +0x1b\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x1e887b0?, 0xc000838680?}, 0xc0001d5dd0?)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:233 +0x57\nk8s.io/apimachinery/pkg/util/wait.poll({0x1e887b0, 0xc000838680}, 0x10?, 0xf4a905?, 0xc0009fe570?)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:580 +0x38\nk8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext({0x1e887b0, 0xc000838680}, 0x10?, 0xc000680800?)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:545 +0x49\nk8s.io/apimachinery/pkg/util/wait.PollImmediateUntil(0xc000437dc0?, 0xc000a42480?, 0x0?)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:536 +0x7c\ngithub.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).start(0xc000437dc0, 0x0?)\n\t/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:165 +0x10c\ncreated by github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*ForwardersManager).Register\n\t/workspace/pkg/controller/utils/datadog/forwarders_manager.go:69 +0x365\n"}
29
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
28
	panic: runtime error: invalid memory address or nil pointer dereference
27
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13e5029]
26
25
goroutine 666 [running]:
24
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x73?})
23
	/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/runtime/runtime.go:55 +0xd8
22
panic({0x195e040, 0x2ce1c60})
21
	/usr/local/go/src/runtime/panic.go:838 +0x207
20
github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).getCredentialsV2(0xc0003acc30?, 0xc0003acb00)
19
	/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:611 +0x69
18
github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).setupV2(0xc000437dc0)
17
	/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:261 +0x24f
16
github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).connectToDatadogAPI(0xc000437dc0)
15
	/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:303 +0x58
14
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x18, 0xc000680800})
13
	/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:220 +0x1b
12
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x1e887b0?, 0xc000838680?}, 0xc0001d5dd0?)
11
	/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:233 +0x57
10
k8s.io/apimachinery/pkg/util/wait.poll({0x1e887b0, 0xc000838680}, 0x10?, 0xf4a905?, 0xc0009fe570?)
9
	/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:580 +0x38
8
k8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext({0x1e887b0, 0xc000838680}, 0x10?, 0xc000680800?)
7
	/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:545 +0x49
6
k8s.io/apimachinery/pkg/util/wait.PollImmediateUntil(0xc000437dc0?, 0xc000a42480?, 0x0?)
5
	/go/pkg/mod/k8s.io/apimachinery@v0.23.5/pkg/util/wait/wait.go:536 +0x7c
4
github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*metricsForwarder).start(0xc000437dc0, 0x0?)
3
	/workspace/pkg/controller/utils/datadog/metrics_forwarder.go:165 +0x10c
2
created by github.com/DataDog/datadog-operator/pkg/controller/utils/datadog.(*ForwardersManager).Register
1
	/workspace/pkg/controller/utils/datadog/forwarders_manager.go:69 +0x365

```
